gradle/root/coverage.gradle

```diff
@@ -1,16 +1,19 @@
 apply plugin: 'jacoco'
-
-// TODO later include subprojects, possible ideas https://gist.github.com/aalmiray/e6f54aa4b3803be0bcac
+apply from: 'gradle/dependencies.gradle'
 
 jacoco {
-    toolVersion = "0.7.9"
+    toolVersion = libraries.jacocoVersion
 }
 
-task mockitoCoverage(type: JacocoReport, dependsOn: "test") {
-    executionData files("${buildDir}/jacoco/test.exec")
+task mockitoCoverage(type: JacocoReport, dependsOn: ["test", "kotlinTest:test", "extTest:test", "inline:test"]) {
+    executionData fileTree(project.rootDir.absolutePath).include("**/build/jacoco/*.exec")
```

Comment:
Maybe it would be better to iterate over subprojects and depend on "test" task (by a type or by a name) and use that list to generate also files to include (istead of looking for `*.exec` file)?

It would be even less fragile to have an optional exclusions for specific modules if needed.

Reply:
I tried to make that work, but ran into problems. First of all, not every subproject has this test task. Therefore, I had to somehow filter the correct project. However, if you do something like below Gradle will throw an exception:

```groovy
subprojects {
  if (it.getTaskByName('test')
    dependsOn "${it.path}:test"
}
```

My Groovy is lackluster and I just couldn't get this to work.

Reply:
Let's avoid hardcoding the test task names and coupling with subprojects. 

Something like that should work (please note that I have not tested it :):

```Gradle
task mockitoCoverage(...)

allprojects {
  plugins.withId("java") {
    apply plugin: "jacoco"
    //configure jacoco version, etc.

    mockitoCoverage.dependsOn "test"
    //configure other parts of jacoco plugin or the mockitoCoverage task
  }
}
```